### PR TITLE
Channel based team services

### PR DESCRIPTION
**Pull request by guy93r**

### CODE REVIEW

File: src/accounts/add/add-command.js [hunk 1]
```diff
@@ -1,38 +1,36 @@
...
-const LoggerFactory = require('../core/logging/logger-factory');
-const { getEventMetadata } = require('../core/logging/logging-metadata');
+const Command = require('../../core/commands/command');
+const LoggerFactory = require('../../core/logging/logger-factory');
+const { getEventMetadata } = require('../../core/logging/logging-metadata');
 
 const logger = LoggerFactory.getLogger(__filename);
 
-class SetupCommand extends Command {
+class AddCommand extends Command {
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/add/add-command.js]: 👍

File: src/accounts/add/add-dialog-handler.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            const rawMessage = message.raw_message;
+            const team = rawMessage.team;
+            const user = rawMessage.user;
+
+
+            return this.teamConfigService.addAccount(team.id, config)
+              .then(() => {
+                bot.reply(message, 'Configuration saved!', err => {
+                  if (!err && message.callback_id === 'initialization_setup_dialog') {
+                    bot.reply(message, `Seems like this is the first configured account, to set as default account just type <@${bot.identity.id}> set workspace account ${alias}`);
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/add/add-dialog-handler.js]: 👍

File: src/accounts/add/add-dialog-handler.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
+const HttpMethod = require('../../core/client/http-method');
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/add/add-dialog-handler.js]: 👍

File: src/accounts/add/add-dialog-handler.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        });
+    });
+  }
+
+  getRealName(token, region) {
+    this.httpClient.getRealName(token, region)
+  }
+}
+
+module.exports = AddDialogHandler;
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/add/add-dialog-handler.js]: 👍

File: src/accounts/add/add-dialog-sender.js [hunk 5]
```diff
@@ -111,4 +112,4 @@ class SetupDialogSender {
 
 }
 
-module.exports = SetupDialogSender;
+module.exports = AddDialogSender;
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/add/add-dialog-sender.js]: 👍

File: src/accounts/channel/clear-active-command.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    });
+  }
+
+  getCategory() {
+    return 'channel configuration';
+  }
+
+  getUsage() {
+    return [
+      '*clear active account {account alias}* - clear account for channel.',
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/channel/clear-active-command.js]: 👍

File: src/accounts/channel/clear-active-command.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    ];
+  }
+
+}
+
+module.exports = ClearActiveCommand;
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/channel/clear-active-command.js]: 👍

File: src/accounts/remove/remove-command.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    });
+  }
+
+  getCategory() {
+    return 'configure';
+  }
+
+  getUsage() {
+    return [
+      '*remove account* - remove account.',
```
Comment (imnotashrimp) [inline src/accounts/remove/remove-command.js]: Revise description:
Remove a Logz.io account from Slack

File: src/accounts/channel/set-active-command.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    super();
+    this.channelHandler = channelHandler;
+  }
+  configure(controller) {
+    controller.hears([commandRegex], 'direct_message,direct_mention', (bot, message) => {
+      if (message.type === 'direct_message') {
+        bot.reply(message, "only allowed on channels");
+        return;
+      }
+      let alias = message.text.match(commandRegex)[1];
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/channel/set-active-command.js]: 👍

File: src/accounts/channel/channel-account-handler.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+class ChannelAccountHandler {
+  constructor(teamConfService) {
+    this.teamConfService = teamConfService;
+  }
+
+  setDefault(teamId, channelId, alias) {
+    return this.teamConfService.saveAccountForChannel(channelId, alias, teamId);
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/channel/channel-account-handler.js]: 👍

File: src/accounts/default/clear-default-command.js [hunk 11]
```diff
@@ -0,0 +1,28 @@
+const Command = require('../../core/commands/command');
+const commandRegex = /clear workspace account/;
+
+class ClearDefault extends Command {
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/default/clear-default-command.js]: 👍

File: src/accounts/default/set-default-command.js [hunk 12]
```diff
@@ -0,0 +1,28 @@
+const Command = require('../../core/commands/command');
+const commandRegex = /set workspace account `(.*)`/;
+
+class SetDefault extends Command {
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/default/set-default-command.js]: 👍

File: src/accounts/default/set-default-command.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+
+  constructor(defaultHandler) {
+    super();
+    this.defaultHandler = defaultHandler;
+  }
+  configure(controller) {
+    controller.hears([commandRegex], 'direct_message,direct_mention', (bot, message) => {
+      let alias = message.text.match(commandRegex)[0];
+      this.defaultHandler.setDefault(message.team, alias);
+      bot.reply(message, `account ${alias} was set as default`)
```
[the remark this answers is not in the document]
Comment (guy93r) [reply src/accounts/default/set-default-command.js]: > `Account ${alias} was set as default account for workspace`

👍

**Pull request closed without merging**